source/Octostache/Templates/Functions/TextSubstringFunction.cs

```diff
@@ -0,0 +1,23 @@
+using System.Linq;
+
+namespace Octostache.Templates.Functions
+{
+    internal static class TextSubstringFunction
+    {
+        public static string Substring(string argument, string[] options)
+        {
+            if (argument == null ||
+                options.Length == 0 ||
+                options.Any(o => !int.TryParse(o, out int _)) ||
+                options.Any(o => int.Parse(o) < 0) ||
+                (options.Length == 1 && int.Parse(options[0]) > argument.Length) ||
+                (options.Length == 2 && (int.Parse(options[0]) + int.Parse(options[1]) > argument.Length)))
+                return null;
+
+            return argument.Substring(
+                options.Length == 1 ? 0 : int.Parse(options[0]),
+                options.Length == 1 ? int.Parse(options[0]) : int.Parse(options[1])
+            ).Trim();
```

Comment:
I'm not sure about the `Trim()`. As a caller, would I expect to get back exactly the section I asked for, not the trimmed version of that. Maybe make it an option? Maybe a Trim function they could pipe the result through if they want if trimmed?

Reply:
Yea, I agree